### PR TITLE
doc: Add quickstart document

### PR DESCRIPTION
**Pull request by cfergeau**

### CODE REVIEW

**Comment by cfergeau** [comment]
Direct link to the quickstart document https://github.com/cfergeau/vfkit/blob/quickstart/doc/quickstart.md
I need to fix a few links.

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md:107]: There were issues with the 2 first links in the doc, but I've fixed them right after filing the PR. 
I'm not sure what's wrong with the gpu link though? It's working for me?

Comment (anjannath) [reply doc/quickstart.md:107]: It says `GUI support`, and there's this section about GUI: https://github.com/crc-org/vfkit/blob/main/doc/usage.md#enabling-a-graphical-user-interface i thought maybe you wanted to link to that

Comment (cfergeau) [reply doc/quickstart.md:107]: Ah yes definitely, I did not even remember this section :p 
I'll change the link, thanks!

**Review by anjannath — APPROVED** [review]

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:28]: This sentence is confusing. I believe the comma before `qcow` should be a full stop instead ,as both are not supported, right?

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:40]: why not `mv` ? the resulting copy is only to use a consistent name in the next paragraphs.

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:43]: this seems unnecessary information, as you explain what a 'copy' of a virtual disk image represents. It would most likely be easier to just call the copy in that case simply something like: `fedora-vfkit-test.raw` or whatever.

Comment (gbraad) [reply doc/quickstart.md:43]: you do not include a cleanup for this?

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:60]: `no logs from the vm` => you mean terminal output?

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:90]: fedora or Fedora?

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:103]: 'This is it for this initial kickstart guide.' not providing any information; can be left out.

**Review by gbraad — CHANGES_REQUESTED** [review]
several comments.
 - `cp` needed, as you use aan unpacked image anyway. might make the name clearly indicate a test, and cleanup
 - fedora or Fedora?
 - other small suggestions

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:5]: `command line-based` ?

Comment (gbraad) [reply doc/quickstart.md:5]: `vfkit is a command line-based hypervisor for macOS.`

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:6]: and continue the sentence with
`, which uses ...`

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:9]: anything about OS requirements?

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md:40]: It's a good point that `cp -c` might be "too advanced" for this guide. However, to me it's a very neat feature to be aware of, it's macOS-specific so users are likely not to be aware of it, and it allows to have the equivalent of qcow2 backing files. You keep a base image, and you have an overlay with your changes on top of it. Maybe move this to its own paragraph with more details? Or to the reference documentation ?

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md:43]: No cleanup for the downloaded file either, I would leave that up to the reader?

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md:40]: I have left this unchanged in the update I just made, but will do more changes now.

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md]: Yes, I've changed it a little bit.

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md]: I changed to `commandline-based` hypervisor, I think I use "commandline" in a single word in other places. Looks like it should be `command-line` though https://en.wikipedia.org/wiki/Command-line_interface , I'll update this!

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md]: Done.

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md]: I added some OS requirements, good point!

**Comment by cfergeau** [comment]
One big unsanswered question in this document is how to inject a ssh key into the image, or get shell access to the guest, or do anything useful with the guest. This is an image which uses cloud-init, and I don't know how to use this with vfkit, so it will stay unanswered for now.

File: doc/quickstart.md
Comment (cfergeau) [reply doc/quickstart.md]: I've addressed this now, only using `mv` in the quick start document, and pointing at a more detailed doc about `cp -c` and `truncate` and corresponding syscalls.

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:9]: some features? if EFI is not available `< 13` it might be wise to explain this.

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:28]: `Please note that qcow` to avoid starting the sentence without a capital.

File: doc/quickstart.md
Comment (gbraad) [inline doc/quickstart.md:64]: I would suggest to use `linux` bootloader as a regular part of sentence and moving the `usage.md` to the end of the following sentence.

`This requires a .... Details about the use can be found in the usage instructions.`

File: doc/usage.md
Comment (gbraad) [inline doc/usage.md:125]: 👍

**Review by gbraad — CHANGES_REQUESTED** [review]
Small suggestions ...
Overall LGTM, so could be merged.

**Comment by cfergeau** [comment]
> Small suggestions ... Overall LGTM, so could be merged.

Updated, I'll merge it tomorrow if no more comments.

**Review by gbraad — APPROVED** [review]

**Pull request merged**
